fix(client): make search bar only highlight footer when it is selectd.

ojeytonwilliams opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or Gitpod.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

This is loosely related to getting to be able to type check the client in CI. On main, the test is upset that `t` is not passed to the SearchBar. Rather than figuring that out, it seemed easier to simply migrate to a functional component. That needed to happen sooner or later.

Edit: this was initially just a refactor, but I ended up fixing a bug that Tom spotted.

<!-- Feel free to add any additional description of changes below this line -->

ojeytonwilliams committed:
refactor(client): make search bar functional

ojeytonwilliams committed:
fix: prop types

moT01 reviewed (commented):
The arrow buttons to go through the results don't seem to be working for me:

<details><summary>gif of this PR</summary>

![search-bar](https://github.com/user-attachments/assets/12504360-2a2b-4a25-93f9-7f8d3731641e)

^ I am pushing down a bunch, highlighted item not changing. Same on firefox and chrome.

</details>

Did find something a little strange on main as well:

<details><summary>gif of main</summary>

![search-bar-main](https://github.com/user-attachments/assets/ba77deec-b444-4de5-b211-dbd633497bbc)

^ It highlights the "See all results" when I get to one of the articles in the middle. Hitting enter goes to the article. Also, when I get to the bottom, it does not highlight the "See all results" if I try to go to it with the array key.

</details>

ojeytonwilliams committed:
fix: make sure index updates on keypress

Sembauke commented:
Works good now!

ojeytonwilliams committed:
fix: select footer when reached

Previously it was hard-coded to select it when the 6th element was
reached via the keyboard.

ojeytonwilliams commented:
Thanks, Sem. I pushed one more commit to fix the second issue Tom spotted. I've adjusted the title to reflect that.
